common/mqtt_client/src/lib.rs

```diff
@@ -269,12 +293,16 @@ impl MqttClient for Client {
             .await
             .map_err(|err| MqttClientError::ClientError(rumqttc::ClientError::Request(err)))?;
 
+        // Track number of pending publish requests.
+        self.pending_publish_count.fetch_add(1, Ordering::Relaxed);
         match qos {
-            // Track number of pending publish requests in case of QoS=1 and 2.
-            QoS::AtLeastOnce | QoS::ExactlyOnce => {
-                self.pending_published_count.fetch_add(1, Ordering::Relaxed);
+            QoS::AtLeastOnce => {}
```

Comment:
why there is no count increment here?, but there was in the earlier case and it was working fine.

Reply:
With the current fix:
* `pending_publish_count` is incremented on `publish` for *all* messages and decremented on `Outgoing(Publish)` i.e. when actually published.
* `pending_puback_count` is incremented for QoS 1, hence decremented on PubAck.
* `pending_pubcomp_count` is incremented for QoS 2, hence decremented on PubComp i.e. the completion of level 2 exchange.